syntax = "proto3";
package tradingcore2pb;

// TrainData - train data
message TrainData {
    int64 trainID = 1;
    string exchangeName = 2;
    string assetsName = 3;
    string strategyName = 4;
    string strParam = 5;
}

// TrainResult - train result
message TrainResult {
    int64 trainID = 1;
    float maxDrawdown = 2;
    float sharpe = 3;
    float annualizedReturns = 4;
    float annualizedVolatility = 5;
    float totalReturns = 6;
    int32 tradingTimes = 7;
    int32 failTimes = 8;
    int32 stoplossTimes = 9;
    float winRate = 10;
    string exchangeName = 11;
    string assetsName = 12;
    string strategyName = 13;
    string strParam = 14;
}

// TradingCore2Service - TradingCore2 service
service TradingCore2Service {
    // train - train
    rpc train(TrainData) returns (TrainResult) {}
}